src/hooks/shared/AccountKeyHashLib.sol

```diff
@@ -0,0 +1,11 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.23;
+
+type AccountKeyHash is bytes32;
+
+library AccountKeyHashLib {
+    /// @notice Rehash a key hash with the sender's account address
+    function wrap(bytes32 keyHash) internal view returns (AccountKeyHash) {
+        return AccountKeyHash.wrap(keccak256(abi.encode(msg.sender, keyHash)));
```

Comment:
I think to make this more extensible it would be best to add the address as an input, then we can see from the base contract what this is wrapped with ?

Reply:
yes good idea!